### ISSUE TITLE
Format for submitted links

### ISSUE THREAD

**Issue by onebree** [EbookFoundation/free-science-books#10]
Unlike programming books, science references have been around for decades, with some subjects rarely updating (math). I think the following should be included when submitting a link
- Title of work/reference
- Format of reference (PDF, HTML, Repository, PPT), unless it is a course
- Author of reference
  + includes PDF, HTML, Repos
  + Excludes wiki-based material (like wikibooks)
- Year published -- excludes constantly modified works (Wikis, Repos)

I am not sure on the exact format. So far, it looks like @eshellman is formatting it as:

    <title> <format> by <author>.

The programming repo does not form it into a sentence, and replaces "by" with " - ".

Any changes we decide here, I will put in my future PR that edits the contributing file.

**Comment by eshellman** [comment]
Personally I like indications when the target of a link is pdf or epub, i.e. not a web page. Also notations when there's no download - as in [online book]. Also, should make clear that affiliate links are not acceptable. 

Also look at: https://sn.reddit.com/r/freeEBOOKS/search?q=flair%3AScience&restrict_sr=on&sort=new&t=all#science

**Comment by onebree** [comment]
Yes, we can assume that references are HTML/webpages, unless otherwise specified. I think we should also note whether accounts are required (like how online courses work). What do you mean by affiliate link? I do not agree with personal accounts like Dropbox, Google Drive, or even torrents.

Maybe we can have a key at the end, full of superscript endnotes? Or even an emoji picture (if there is one)

**Comment by eshellman** [comment]
affiliate links are those that direct revenue to the linker, i.e. https://affiliate-program.amazon.com/

**Comment by onebree** [comment]
Okay, makes sense... Could you give me an example of a free ebook that is from an affiliate link?

**Comment by eshellman** [comment]
http://www.amazon.com/gp/product/B00XJHMDZG/ref=as_li_tl?ie=UTF8&camp=1789&creative=390957&creativeASIN=B00XJHMDZG&linkCode=as2&tag=gotohe-20&linkId=XZZAWBX6XW46DOPW

**Comment by onebree** [comment]
Oh, I understand. I know FPB recently added a book that Microsoft released. What about something like that?

[Microsoft Press: Free E-Books](http://blogs.msdn.com/b/microsoft_press/archive/2011/03/03/ebooks-list-of-our-free-books.aspx)

**Comment by eshellman** [comment]
The microsoft press books are fine

**Comment by onebree** [comment]
Okay, I was just making sure... I agree with NO affiliate links then. You have any ideas on a final format per list item?

**Comment by eshellman** [comment]
Right now, a "final format" would be a premature optimization

**Comment by fabiohenriquecn** [comment]
Could we use this format:

* [Wisconsin Calculus](https://www.math.wisc.edu/undergraduate/calcstudentpage) (PDF) by Sigurd B. Angenent et al

for books with 4 authors?

And the book with 2 authors or more?

**Comment by eshellman** [comment]
sure, "et al."

**Comment by fabiohenriquecn** [comment]
All the "." at the end of line could be removed.

**Comment by eshellman** [comment]
I agree. I will add a linter and travis ci config.

**Comment by fabiohenriquecn** [comment]
Can I add this :

* [Mathematics For Computer Science](https://courses.csail.mit.edu/6.042/spring17/mcs.pdf), by Eric Lehman et al.

?

It's the best link for this book that I found.

**Comment by eshellman** [comment]
sure. I'm surprised it's not in free programming books.
